### PR TITLE
Add privileges for ES

### LINKED ISSUE #3224: Document ElasticSearch required privileges for SM
## Summary

Some customers need to know which privileges C8 applications requires when accessing a locally set-up ES. This allows them to fine-tune the permissions they grant. 

## Problem

Currently, this list of privileges is not discoverable in the docs. We have ad-hoc collected this list (see https://camunda.slack.com/archives/C026U8GBNSW/p1705485613567079?thread_ts=1705068753.577989&cid=C026U8GBNSW), but should document this in a central place discoverable for our users.

### PR DESCRIPTION
**Pull request by akeller**
## Description

@romansmirnov how far can I backport this change? All supported versions? 

Closes #3224. 

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/self-managed/operational-guides/configure-elasticsearch-privileges.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+## Cluster privileges
+
+To use the [backup feature](/self-managed/operational-guides/backup-restore/backup-and-restore.md), you need to have snapshot privileges:
+
+- `create_snapshot`
+- `monitor_snapshot`
+- `manage_index_templates` to manage index templates
+- `monitor`
+- `manage_ilm` when ILM is enabled
```
Comment (romansmirnov) [inline docs/self-managed/operational-guides/configure-elasticsearch-privileges.md:16]: These three are not required for taking a backup.

* `monitor` is necessary for the health check
* `manage_index_templaces` are kind of to create the index schema on startup (if they don't exist yet in Elasticsearch)
* `manage_ilm` when the index lifecycle management is used to apply a retention policy


I think I would mark the first (monitor) and the second manage_index_templaces privilege as required. The third only when they apply ILM.

**Review by romansmirnov — CHANGES_REQUESTED** [review]
Thanks for opening the PR. I have added a comment on the cluster privileges.

**Comment by akeller** [comment]
@christinaausley I realized we won't have info in here about configuring ES privileges, so it feels weird to place this file in the **Operational guides** section. Do you have another suggestion? 

For additional context, ES is used by multiple components in Self-Managed. This could be a reference topic, but it's not for SaaS users.

**Comment by akeller** [comment]
> Could we not fit this into Concepts? I feel like we've been having a handful of "Where does this misc doc go in SM?" conversations. Could always launch a sidebar section of additional resources that apply to all Self-Managed components, but IMO this is the goal of Concepts, no?

Yes! What a great idea. I was too stuck in thinking this was config and, therefore, must be a guide.

File: versioned_docs/version-8.1/self-managed/concepts/elasticsearch-privileges.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+title: "Elasticsearch privileges"
+---
+
+If you implement Camunda 8 with Elasticsearch as a service provider, you must configure Elasticsearch with the following [privileges](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-privileges.html) in mind:
+
+## Cluster privileges
+
+- `monitor` - necessary for health check
+- `manage_index_templates` to create and manage index schema on start up, if they don't already exist in Elasticsearch.
+- _Optional_ `manage_ilm` - required only when ILM is enabled
```
Comment (romansmirnov) [inline versioned_docs/version-8.1/self-managed/concepts/elasticsearch-privileges.md:12]:
```suggestion
```

Not required with 8.1.

File: versioned_docs/version-8.1/self-managed/concepts/elasticsearch-privileges.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+More information on cluster privileges in Elasticsearch can be found [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-privileges.html#privileges-list-cluster).
+
+## Indices privileges
+
+- `create_index`
+- `delete_index`
+- `read`
+- `write`
+- `manage`
+- _Optional_ `manage_ilm` - required only when ILM is enabled
```
Comment (romansmirnov) [inline versioned_docs/version-8.1/self-managed/concepts/elasticsearch-privileges.md:32]:
```suggestion
```
Not required with 8.1.

File: versioned_docs/version-8.2/self-managed/concepts/elasticsearch-privileges.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+title: "Elasticsearch privileges"
+---
+
+If you implement Camunda 8 with Elasticsearch as a service provider, you must configure Elasticsearch with the following [privileges](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-privileges.html) in mind:
+
+## Cluster privileges
+
+- `monitor` - necessary for health check
+- `manage_index_templates` to create and manage index schema on start up, if they don't already exist in Elasticsearch.
+- _Optional_ `manage_ilm` - required only when ILM is enabled
```
Comment (romansmirnov) [inline versioned_docs/version-8.2/self-managed/concepts/elasticsearch-privileges.md:12]:
```suggestion
```
Not required with 8.2.

File: versioned_docs/version-8.2/self-managed/concepts/elasticsearch-privileges.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+More information on cluster privileges in Elasticsearch can be found [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-privileges.html#privileges-list-cluster).
+
+## Indices privileges
+
+- `create_index`
+- `delete_index`
+- `read`
+- `write`
+- `manage`
+- _Optional_ `manage_ilm` - required only when ILM is enabled
```
Comment (romansmirnov) [inline versioned_docs/version-8.2/self-managed/concepts/elasticsearch-privileges.md:32]:
```suggestion
```
Not required with 8.2.

**Review by romansmirnov — DISMISSED** [review]
@akeller, LGTM! I believe the ILM privileges are not required with 8.1 and 8.2 - I went through the change logs to prove.

File: versioned_docs/version-8.2/self-managed/concepts/elasticsearch-privileges.md (hunk 5 above)
Comment (romansmirnov) [reply versioned_docs/version-8.2/self-managed/concepts/elasticsearch-privileges.md:32]: I was wrong! It is available with 8.2

File: versioned_docs/version-8.2/self-managed/concepts/elasticsearch-privileges.md (hunk 4 above)
Comment (romansmirnov) [reply versioned_docs/version-8.2/self-managed/concepts/elasticsearch-privileges.md:12]: I was wrong! It is available with 8.2

**Comment by akeller** [comment]
@christinaausley can I get a review from you and a 👍 so I can merge?

**Review by christinaausley — APPROVED** [review]

**Pull request merged**